### DISCUSSION TITLE
Getting an error while trying to download app from chrome browser

### DISCUSSION THREAD

**Discussion by dnbucket88** [codemagic-ci-cd/codemagic-docs#2290] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

Hello friends, i have created a java code which converts website into app. When i use it and try to download an app file like .aab or .apk from firefox it works correctly but when i am trying to downloading it from chrome it shows this error " {"error":"FORBIDDEN","message":"The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required."} "

This  is a link of a build - https://api.codemagic.io/artifacts/8ee9e63e-98a5-4b3e-8ba4-e5660d6a1f79/880584fe-2b70-4569-b03d-864935b8b7a0/app-release.aab

### Expected results

My expected result is it should work correctly like firefox

### Actual results

{"error":"FORBIDDEN","message":"The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required."}

**Comment by dnbucket88** [comment]
Anyone ? Not able to download from mobile also

**Reply by dtrdic** [reply]
Hey @dnbucket88, so you're saying that you can download the binary in Firefox but not in Chrome? Could you check if you have any browser extensions that might be interfering with the download? Try disabling extensions or add-ons and see if that resolves the issue.
Try also clearing cache and cookies

**Comment by dnbucket88** [comment]
Sir i tried but its not working on chrome as well as on mobile and on Microsoft Edge.

**Reply by dtrdic** [reply]
I suspect that you might be connected to the wrong account on other browsers. Have you tried logging in to the Codemagic platform directly from those browsers? Or are you simply copy-pasting the above link into different browsers?
Please also share the build id which can be found on the build overview page
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/81fae12e-7806-47db-b752-6573dd397868)

**Reply by dtrdic** [reply]
Also, if you intend to share build artifacts outside of Codemagic, you'll need to create a shared build dashboard. 
Direct links to artifact downloads, like the one you shared above, will not work for this purpose. Please check out here on how to enable build dashboards [here](https://docs.codemagic.io/yaml-publishing/build-dashboards/).

**Reply by dnbucket88** [reply]
As I can see we need to share the whole dashboard and it doesnt support api

**Reply by dtrdic** [reply]
You can also try creating a public download URL from the one you shared above, check [this out](https://docs.codemagic.io/rest-api/artifacts/#step-2-create-a-public-download-url-using-the-url-obtained-in-step-1)

**Reply by dnbucket88** [reply]
Sir Can you suggest edit https://github.com/dnbucket88/websiteintoapp/blob/a845cbb5cb936e5a234c7741bec813454fafdb13/codemagic.yaml so that i can create public url

**Reply by dtrdic** [reply, marked as answer]
you don't have to edit your yaml, just make a POST request in postman with the URL of the artifact you just provided, and send `{"expiresAt": 1785419345} `as a JSON payload in the body

<img width="952" alt="Screenshot 2024-01-05 at 13 43 10" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/e582c2e1-ab1d-44c5-b528-38d0ebede8e1">

then you will get the the response that contains the public artifact download URL under the url key, which you can share

**Comment by dnbucket88** [comment]
Sir, i am copy pasting link. I have built many apps but for share purposes. U can check this one 

Build overview
ID:
6593eb126b02b630fb7c9b3b
Index:
37

**Comment by dtrdic** [comment]
I am closing the thread due to inactivity. Let me know if you have any more questions by replying to this thread or opening a new one.
